Adding xVMNetworkAdapter resource

rchaganti opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xhyper-v/53)

<!-- Reviewable:end -->

msftclas commented:
Hi __@rchaganti__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

rchaganti commented:
Any idea when this will get reviewed or merged?

iainbrighton commented:
Review status: 0 of 8 files reviewed at latest revision, 7 unresolved discussions.

---

*[README.md, line 77 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbJfVXU6q1wh5UfsiO:-KIbJfVXU6q1wh5UfsiP:-2095566002) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/README.md#L77)):*
> ```Markdown
> 
> * **Id**: Unique string for identifying the resource instance.
> * **Name**: Name of the network adapter as it appears either in the management OS or attached to a VM.
> ```

How is the Name parameter different from the Id parameter? Is it possible or desirable to have the same name for multiple adapters, for example, when teaming?

---

*[README.md, line 79 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbHpgwjzxis1ekTXG9:-KIbHpgwjzxis1ekTXGA:321975917) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/README.md#L79)):*
> ```Markdown
> * **Name**: Name of the network adapter as it appears either in the management OS or attached to a VM.
> * **SwitchName**: Virtual Switch name to connect the adapter to.
> * **VMName**: Name of the VM to attach to. If you want to attach new VM Network adapter to the management OS, set this property to 'Management OS'.
> ```

Should this property name have a space in it? It might lead to confusion.

---

*[README.md, line 80 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbHxpVhjQsFueHv8GM:-KIbHxpVhjQsFueHv8GN:607918992) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/README.md#L80)):*
> ```Markdown
> * **SwitchName**: Virtual Switch name to connect the adapter to.
> * **VMName**: Name of the VM to attach to. If you want to attach new VM Network adapter to the management OS, set this property to 'Management OS'.
> * **DynamicMacAddress**: Set this to $false if you want to specify a static MAC address.
> ```

Could the Dynamic/Static be inferred? For example, if no MacAddress specified, default to dynamic? This would remove an unnecessary parameter?

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 143 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbKfbjRh4b7diSzfAx:-KIbKfbjRh4b7diSzfAy:-1427880144) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L143)):*
> ```PowerShell
>                     }
>                 }
>                 elseif ($NetAdapterExists.SwitchName -ne $SwitchName)
> ```

Should this not just be an 'if' statement as it will only currently trigger if the Mac address is correctly configured? If the Mac address setting is configured incorrectly, it looks like it would take 2 consistency passes to converge.

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 264 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbM5Iuzj0njJosbGcr:-KIbM5Iuzj0njJosbGcs:-462597626) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L264)):*
> ```PowerShell
>                     }
>                 }
>                 elseif ($netAdapterExists.SwitchName -ne $SwitchName)
> ```

I don't think this will trigger as both $DynamicMacAddress and $StaticMacAddress cannot be $null? Should just be an 'if'?

---

*[Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1, line 117 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbNi9tc1QkOpChV1wq:-KIbNi9ujndHioaUdDJD:1575497040) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1#L117)):*
> ```PowerShell
>                 }
>             }
> ```

There are no tests for the $SwitchName/Connect-VMNetAdapter or updating/Set-VMNetAdapter calls (if you update the tests first, you should see the logic issues mentioned earlier highlighted).

---

*[Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1, line 175 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbTEGMM4rQP5NL1TAt:-KIbTEGMM4rQP5NL1TAu:614876002) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1#L175)):*
> ```PowerShell
>                 }
>             }
> ```

Missing tests for determining whether DynamicMacAddress/StaticMacAddress, VMName and SwitchName are correct/incorrect (if you update the tests first, you should see the logic issues mentioned earlier highlighted).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
All good questions. I am still not use to reviewable so I will respond here.

#1. A net adapter connected to either VM or management OS can have the same name as others. So, it will be impossible to add multiple adapters with the same name even to different VMs if Adapter name is the key property. THis is where Id is being used. Refer to https://github.com/rchaganti/DSCResources/issues/3

#2. Agree with you. I can change it to ManagementOS.

#3. Sure, makes sense. 

I am still reviewing #4, #5, #6, and #7. I will let you know my feedback soon.

zjalali commented:
Review status: 0 of 8 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 143 \[r2\]](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbKfbjRh4b7diSzfAx:-KOM_iYES1LVHSClrR6T:-1254022819) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L143)):*
<details><summary><i>Previously, iainbrighton (Iain Brighton) wrote…</i></summary>
> Should this not just be an 'if' statement as it will only currently trigger if the Mac address is correctly configured? If the Mac address setting is configured incorrectly, it looks like it would take 2 consistency passes to converge.
</details>
Any  updates on this?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@mbreakey3, @rchaganti - do we want to make sure all new PR's to this module meet HQRM guidelines? There are a few minor bits that need should be addressed in this PR if we're wanting to meet HQRM. I'm happy to do a review with those guidelines in mind (I can't remember the old guidelines now so I can only do HQRM reviews :smile: )

But no big deal if not - I'll be working on getting this module to HQRM standards over the next few months anyway and will address any problems at that point. But it would help if there were less things to fix up :grin:

rchaganti commented:
@PlagueHO I am trying to push all updates to my repo first. Let us review the code there and then I will submit a PR directly to the HQRM module. I have a couple of new resources as well in cHyper-V. So, I am trying to address the HQRM guidelines one step at a time in that repo.

PlagueHO commented:
@rchaganti - Ok cool! I'll leave the HQRM stuff for a bit on this repos. Will look at addressing it once you've merged all your modules in.

mbreakey3 commented:
Since this isn't a whole new module submission, it does _not_ need to meet all of the HQRM guidelines to be merged. Tests are required, general high code quality is required but it doesn't need to follow the strict HQRM rules yet.

rchaganti commented:
OK! I am updating the tests to give some comprehensive unit test coverage. I will push the module updates this week.

rchaganti commented:
Alright! I have incorporated all review feedback and updated the tests / examples to reflect new parameters. This build is passing. So, please review the code and let me know if there is anything that needs further change.

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[README.md, line 77 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbJfVXU6q1wh5UfsiO:-KV9u2FN0Y18na8_7kwC:bulhqdr) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/README.md#L77)):*
<details><summary><i>Previously, iainbrighton (Iain Brighton) wrote…</i></summary>
> How is the Name parameter different from the Id parameter? Is it possible or desirable to have the same name for multiple adapters, for example, when teaming?
</details>
Id ensures that there can be a VM adapter with the same name in multiple VMs. Without this unique key and Name being the key, we won't be able to have multiple resource instances even though they belong to different VMs. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[README.md, line 79 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbHpgwjzxis1ekTXG9:-KV9ufouK8xQX5dqV5Zz:b-xqqhd3) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/README.md#L79)):*
<details><summary><i>Previously, iainbrighton (Iain Brighton) wrote…</i></summary>
> Should this property name have a space in it? It might lead to confusion.
</details>
I removed the space. This will be now ManagementOS.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[README.md, line 80 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbHxpVhjQsFueHv8GM:-KV9uocYYQ7HDnpcr1DD:b7eceom) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/README.md#L80)):*
<details><summary><i>Previously, iainbrighton (Iain Brighton) wrote…</i></summary>
> Could the Dynamic/Static be inferred? For example, if no MacAddress specified, default to dynamic? This would remove an unnecessary parameter?
</details>
Yes, this input made sense to me. There is only MacAddress property now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 143 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbKfbjRh4b7diSzfAx:-KV9uxZltb6QQGXo3yic:bh38azu) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L143)):*
> his?
Done! there as a logic error here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 264 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbM5Iuzj0njJosbGcr:-KV9v0KPxR8CHDKgXcY-:b-u4r77i) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L264)):*
<details><summary><i>Previously, iainbrighton (Iain Brighton) wrote…</i></summary>
> I don't think this will trigger as both $DynamicMacAddress and $StaticMacAddress cannot be $null? Should just be an 'if'?
</details>
Fixed. There are no two parameters now. They are combined into MacAddress and we infer the need for static or dynamic from that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1, line 175 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbTEGMM4rQP5NL1TAt:-KV9vEXEF402jB1ENSe6:bx0406p) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1#L175)):*
<details><summary><i>Previously, iainbrighton (Iain Brighton) wrote…</i></summary>
> Missing tests for determining whether DynamicMacAddress/StaticMacAddress, VMName and SwitchName are correct/incorrect (if you update the tests first, you should see the logic issues mentioned earlier highlighted).
</details>
These are not needed anymore

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1, line 117 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KIbNi9tc1QkOpChV1wq:-KV9vI4Shw-LC6A1gIXE:b-63o8gl) ([raw file](https://github.com/powershell/xhyper-v/blob/d6708d15c5d802f5fa1723bed99984afc76d965b/Tests/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.Tests.ps1#L117)):*
<details><summary><i>Previously, iainbrighton (Iain Brighton) wrote…</i></summary>
> There are no tests for the $SwitchName/Connect-VMNetAdapter or updating/Set-VMNetAdapter calls (if you update the tests first, you should see the logic issues mentioned earlier highlighted).
</details>
This is mostly not needed. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwiFLplyyP_4u9v3f:-KVPwiFLplyyP_4u9v3g:buny172) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1#L3)):*
> ```PowerShell
> Configuration HostOSAdapter
> {
>     Import-DscResource -ModuleName cHyper-V -Name cVMNetworkAdapter
> ```

Should this be "xVMNetworkAdapter"?

---

*[Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwj5zhXwKMsIVbPUp:-KVPwj5zhXwKMsIVbPUq:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1#L6)):*
> ```PowerShell
>     Import-DscResource -ModuleName PSDesiredStateConfiguration
> 
>     cVMNetworkAdapter HostOSAdapter {
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwXMsw9um0Lys2T9j:-KVPwXMsw9um0Lys2T9k:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1#L6)):*
> ```PowerShell
>     Import-DscResource -ModuleName PSDesiredStateConfiguration
> 
>     cVMNetworkAdapter ManagementAdapter {
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1, line 14 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwbNsaF09rxi9TiV1:-KVPwbNsaF09rxi9TiV2:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1#L14)):*
> ```PowerShell
>     }
> 
>     cVMNetworkAdapter ClusterAdapter {
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwm1WyYgM7eFcgCHn:-KVPwm1WyYgM7eFcgCHo:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L3)):*
> ```PowerShell
> Configuration VMAdapter
> {
>     Import-DscResource -ModuleName cHyper-V -Name cVMNetworkAdapter
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwoOXD5GnuTHsrlVk:-KVPwoOXD5GnuTHsrlVl:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L6)):*
> ```PowerShell
>     Import-DscResource -ModuleName PSDesiredStateConfiguration
> 
>     cVMNetworkAdapter MyVM01NIC {
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 14 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwpXBt2r8WOxk14hl:-KVPwpXBt2r8WOxk14hm:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L14)):*
> ```PowerShell
>     }
> 
>     cVMNetworkAdapter MyVM02NIC {
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 22 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwr6ySzZ5L0v_aiUz:-KVPwr6ySzZ5L0v_aiV-:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L22)):*
> ```PowerShell
>     }
> 
>     cVMNetworkAdapter MyVM03NIC {
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwt5sRsXYWYLwFFCe:-KVPwt5sRsXYWYLwFFCf:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1#L3)):*
> ```PowerShell
> Configuration VMAdapter
> {
>     Import-DscResource -ModuleName cHyper-V -Name cVMNetworkAdapter
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwuG3r54jadhyhhhY:-KVPwuG7JGC03sgibnJ6:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1#L6)):*
> ```PowerShell
>     Import-DscResource -ModuleName PSDesiredStateConfiguration
> 
>     cVMNetworkAdapter MyVM01NIC {
> ```

"c" resource

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1, line 15 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwvlycqaDAKRiSWDy:-KVPwvlycqaDAKRiSWDz:b-6mnoby) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1#L15)):*
> ```PowerShell
>     }
> 
>     cVMNetworkAdapter MyVM02NIC {
> ```

"c" resource

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
*facepalm* :) I will fix all that.

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwiFLplyyP_4u9v3f:-KVSxMBCjePXqlBRWXko:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1#L3)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Should this be "xVMNetworkAdapter"?
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwj5zhXwKMsIVbPUp:-KVSxR_wdg7n8LwbuqZu:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_ManagementOS.ps1#L6)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwXMsw9um0Lys2T9j:-KVSxXd4T6WH99r6lVa_:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1#L6)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1, line 14 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwbNsaF09rxi9TiV1:-KVSxZvfsFGdnVjVIipz:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleManagementOS.ps1#L14)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwm1WyYgM7eFcgCHn:-KVSxaf9sz_z2lL-DfUO:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L3)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwoOXD5GnuTHsrlVk:-KVSxbZ4dbXg8dGJBmcl:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L6)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 14 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwpXBt2r8WOxk14hl:-KVSxcq6IkhRhkZfYFMA:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L14)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1, line 22 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwr6ySzZ5L0v_aiUz:-KVSxeA_RlxNiYByPZwg:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVM.ps1#L22)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwt5sRsXYWYLwFFCe:-KVSxfru_dB2McV5IqVx:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1#L3)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1, line 6 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwuG3r54jadhyhhhY:-KVSxgY2Q5CpeDgZPK1p:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1#L6)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 0 of 9 files reviewed at latest revision, 18 unresolved discussions.

---

*[Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1, line 15 at r4](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KVPwvlycqaDAKRiSWDy:-KVSxiZCDVDhPk9ngMLi:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/9a11107505fd9e4f9c0fde3c1ead975d9ca5d5a5/Examples/Sample_xVMNetworkAdapter_MultipleVMMACAddress.ps1#L15)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> "c" resource
</details>
Done!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
@rchaganti - when you're responding to comments on reviewable, can you write all of them and then press the green publish button in the top right of the screen rather than pressing the publish button in the corner of each comment? That way they all get condensed into one comment on gitHub rather than sending out individual messages. Thanks!

rchaganti commented:
Sure! will do. @mbreakey3

iainbrighton commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 9 files reviewed at latest revision, 12 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-:-KVebBaMGzy23KeYJOHy:bnfp4nl)*
<!-- Sent from Reviewable.io -->

iainbrighton commented:
Reviewed 1 of 1 files at r3, 4 of 10 files at r4, 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Reviewed 1 of 1 files at r3, 8 of 10 files at r4.
Review status: all files reviewed at latest revision, 31 unresolved discussions.

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 15 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sVLZyWEJ4LhXI58O:-KW8sVLZyWEJ4LhXI58P:b-fdxfd8) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L15)):*
> ```PowerShell
> #endregion
> 
> Function Get-TargetResource
> ```

Can you add help before each of the functions?

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 81 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sdDIZ5GfLzUo3-_3:-KW8sdDIZ5GfLzUo3-_4:b-fdxfd8) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L81)):*
> ```PowerShell
> }
> 
> Function Set-TargetResource
> ```

Can you add help before each of the functions?

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 119 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sfjfpGzQDAMjkEyC:-KW8sfjfpGzQDAMjkEyD:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L119)):*
> ```PowerShell
>     }
>     
>     Write-Verbose $localizedData.GetVMNetAdapter
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 126 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8shXbC-JMVz7vH-0a:-KW8shXbC-JMVz7vH-0b:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L126)):*
> ```PowerShell
>         if ($netAdapterExists)
>         {
>             Write-Verbose $localizedData.FoundVMNetAdapter
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 153 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8slKQOf93Ff2tRNVX:-KW8slKQOf93Ff2tRNVY:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L153)):*
> ```PowerShell
>                 if ($netAdapterExists.SwitchName -ne $SwitchName)
>                 {
>                     Write-Verbose $localizedData.PerformSwitchConnect
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 159 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8smzr87tM7mvMICkX:-KW8smzr87tM7mvMICkY:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L159)):*
> ```PowerShell
>                 if (($updateMacAddress))
>                 {
>                     Write-Verbose $localizedData.PerformVMNetModify
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 189 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sp2z-iqbnxfJe2Vs:-KW8sp3-AsAi1CanC_hC:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L189)):*
> ```PowerShell
>                 $arguments.Add('SwitchName',$SwitchName)
>             }
>             Write-Verbose $localizedData.AddVMNetAdapter
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 195 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sqOdMJpoqqsCt5CD:-KW8sqOdMJpoqqsCt5CE:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L195)):*
> ```PowerShell
>     else
>     {
>         Write-Verbose $localizedData.RemoveVMNetAdapter
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 200 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8srMn6yLMmDKrL456:-KW8srMn6yLMmDKrL457:b-fdxfd8) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L200)):*
> ```PowerShell
> }
> 
> Function Test-TargetResource
> ```

Can you add help before each of the functions?

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 239 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8ssyz6TH9jGy-DeO-:-KW8ssyz6TH9jGy-DeO0:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L239)):*
> ```PowerShell
>     }
>     
>     Write-Verbose $localizedData.GetVMNetAdapter
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 252 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8svwUPOGE8QCnEQVb:-KW8svwUPOGE8QCnEQVc:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L252)):*
> ```PowerShell
>                     if ($netAdapterExists.DynamicMacAddressEnabled)
>                     {
>                         Write-Verbose $localizedData.EnableStaticMacAddress
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 257 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sxFibkQGvY6fWeHq:-KW8sxFibkQGvY6fWeHr:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L257)):*
> ```PowerShell
>                     elseif ($netAdapterExists.MacAddress -ne $MacAddress)
>                     {
>                         Write-Verbose $localizedData.StaticAddressDoesNotMatch
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 265 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8syfuIMMAzSx8_dHv:-KW8syfuIMMAzSx8_dHw:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L265)):*
> ```PowerShell
>                     if (-not $netAdapterExists.DynamicMacAddressEnabled)
>                     {
>                         Write-Verbose $localizedData.EnableDynamicMacAddress
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 272 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t0dUdh-J7sWJ_GSA:-KW8t0dUdh-J7sWJ_GSB:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L272)):*
> ```PowerShell
>                 if ($netAdapterExists.SwitchName -ne $SwitchName)
>                 {
>                     Write-Verbose $localizedData.SwitchIsDifferent
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 277 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t-JYN82_g8ViS2oL:-KW8t-JYN82_g8ViS2oM:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L277)):*
> ```PowerShell
>                 else
>                 {
>                     Write-Verbose $localizedData.VMNetAdapterExistsNoActionNeeded
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 283 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t2NPoo00UhYr3VLp:-KW8t2NPoo00UhYr3VLq:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L283)):*
> ```PowerShell
>             else
>             {
>                 Write-Verbose $localizedData.VMNetAdapterExistsNoActionNeeded
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 289 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t3QVttS4Get6OaIF:-KW8t3QVttS4Get6OaIG:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L289)):*
> ```PowerShell
>         else
>         {
>             Write-Verbose $localizedData.VMNetAdapterDoesNotExistShouldAdd
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 297 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t4vB4iv5iZSspCI2:-KW8t4vB4iv5iZSspCI3:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L297)):*
> ```PowerShell
>         if ($netAdapterExists)
>         {
>             Write-Verbose $localizedData.VMNetAdapterExistsShouldRemove
> ```

Add named param

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 302 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t64at2xxqIy2phq7:-KW8t64at2xxqIy2phq8:b7542si) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L302)):*
> ```PowerShell
>         else
>         {
>             Write-Verbose $localizedData.VMNetAdapterDoesNotExistNoActionNeeded
> ```

Add named param

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Reviewed 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 31 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
Review status: 8 of 9 files reviewed at latest revision, 31 unresolved discussions.

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 15 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sVLZyWEJ4LhXI58O:-KWYBrVffnxgRaRP796_:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L15)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Can you add help before each of the functions?
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 81 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sdDIZ5GfLzUo3-_3:-KWYBi7enPQLtntPV2l9:b-896fja) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L81)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Can you add help before each of the functions?
</details>
Done!

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 119 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sfjfpGzQDAMjkEyC:-KWYBk4Fq1YRpE4FBdbJ:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L119)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 126 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8shXbC-JMVz7vH-0a:-KWYBuIiag_9Q3lsemDb:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L126)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 153 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8slKQOf93Ff2tRNVX:-KWYBwg0KcVhFk4QJRjA:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L153)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 159 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8smzr87tM7mvMICkX:-KWYBxvBOxFMDMDCm9u6:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L159)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 189 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sp2z-iqbnxfJe2Vs:-KWYBz-ckg_DRafcod0h:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L189)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 195 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sqOdMJpoqqsCt5CD:-KWYBzzb6A63-_t_DpSb:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L195)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 200 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8srMn6yLMmDKrL456:-KWYC09oitMCPt1Qy-OI:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L200)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Can you add help before each of the functions?
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 239 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8ssyz6TH9jGy-DeO-:-KWYC1D6G4QvDxTgIfcF:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L239)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 252 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8svwUPOGE8QCnEQVb:-KWYC2Au6befA1c7MWr0:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L252)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 257 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8sxFibkQGvY6fWeHq:-KWYC3B3DkKJ0bamVPlg:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L257)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 265 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8syfuIMMAzSx8_dHv:-KWYC49OSu_0OFZXhENo:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L265)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 272 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t0dUdh-J7sWJ_GSA:-KWYC5EnhvzCniRp0kkC:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L272)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 277 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t-JYN82_g8ViS2oL:-KWYC6IGfxIOWcgsS_k2:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L277)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 283 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t2NPoo00UhYr3VLp:-KWYC7HD07Di8L8TLStO:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L283)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 289 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t3QVttS4Get6OaIF:-KWYC9LKOWhQgMjqLzYq:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L289)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 297 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t4vB4iv5iZSspCI2:-KWYCAPEMckeYa2sU3Oi:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L297)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---

*[DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1, line 302 at r5](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KW8t64at2xxqIy2phq7:-KWYCBHSq1a92u_tenxU:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/38d393029988682679431043a8f606d435aca38a/DSCResources/MSFT_xVMNetworkAdapter/MSFT_xVMNetworkAdapter.psm1#L302)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary>
> Add named param
</details>
Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
@mgreenegit  Good to go?

rchaganti commented:
Review status: 7 of 9 files reviewed at latest revision, 33 unresolved discussions.

---

*[README.md, line 79 at r6](https://reviewable.io:443/reviews/powershell/xhyper-v/53#-KHXzpaQ0XxW_7ByZH5O-r6-79:-KZPyqWp-j4Px0tTwoXD:bb6un1j) ([raw file](https://github.com/powershell/xhyper-v/blob/b86fd33975c3e567f10fafd486dd4647e7b2c94c/README.md#L79)):*
<details><summary><i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary><blockquote>

Can you note that this is the key?
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/53)*
<!-- Sent from Reviewable.io -->

rchaganti commented:
I guess this one missed the last release. But, just checking. Are we all good here?
